pytorch_lightning/plugins/training_type/deepspeed.py

```diff
@@ -340,6 +341,30 @@ def setup_distributed(self):
             self._format_config()
             self._config_initialized = True
 
+    def init_ddp_connection(self, global_rank: Optional[int] = None, world_size: Optional[int] = None) -> None:
+        if platform.system() != "Windows":
+            # do not set env variables on windows, allow deepspeed to control setup
+            global_rank = global_rank if global_rank is not None else self.cluster_environment.global_rank()
+            world_size = world_size if world_size is not None else self.cluster_environment.world_size()
+            self._set_node_environment_variables(global_rank, world_size)
+            log.info(
+                f"initializing deepspeed distributed: "
+                f"GLOBAL_RANK: {global_rank}, "
+                f"MEMBER: {global_rank + 1}/{world_size}"
+            )
+        deepspeed.init_distributed(
+            self.torch_distributed_backend, distributed_port=self.cluster_environment.master_port()
+        )
+
+    def _set_node_environment_variables(
+        self, global_rank: Optional[int] = None, world_size: Optional[int] = None
+    ) -> None:
+        os.environ["MASTER_ADDR"] = self.cluster_environment.master_address()
+        os.environ["MASTER_PORT"] = str(self.cluster_environment.master_port())
+        os.environ["RANK"] = str(global_rank)
+        os.environ["WORLD_SIZE"] = str(world_size)
+        os.environ["LOCAL_RANK"] = str(self.local_rank)
```

Comment:
noob q: isn't setting these responsibilities of the cluster environment?

Reply:
I think the cluster environments only provide the variables, assuming that others have set it for the environment we're in. In our case we're passing the set variables into a format for DeepSpeed to setup internally